test/replication.test.js

```diff
@@ -1804,3 +1804,105 @@ describe('Replication / Change APIs', function() {
   }
 });
+
+describe('Replication / Change APIs with multi tenancy', function() {
+  this.timeout(10000);
+  var dataSource, SourceModelWithTenant, TargetModelWithTenant;
+  var useSinceFilter;
+  var tid = 0; // per-test unique id used e.g. to build unique model names
+
+  beforeEach(function() {
+    tid++;
+    useSinceFilter = false;
+    var test = this;
+
+    dataSource = this.dataSource = loopback.createDataSource({
+      connector: loopback.Memory
+    });
+    SourceModelWithTenant = this.SourceModelWithTenant = PersistedModel.extend(
+      'SourceModelWithTenant-' + tid,
+      { id: { id: true, type: String, defaultFn: 'guid' } },
+      { trackChanges: true, tenantProperty: 'tenantId' });
+
+    SourceModelWithTenant.attachTo(dataSource);
+
+    TargetModelWithTenant = this.TargetModelWithTenant = PersistedModel.extend(
+      'TargetModelWithTenant-' + tid,
+      { id: { id: true, type: String, defaultFn: 'guid' } },
+      { trackChanges: true, tenantProperty: 'tenantId' });
+
+    var TargetWithTenantChange = TargetModelWithTenant.Change;
+    TargetWithTenantChange.Checkpoint = loopback.Checkpoint.extend('TargetWithTenantCheckpoint');
+    TargetWithTenantChange.Checkpoint.attachTo(dataSource);
+
+    TargetModelWithTenant.attachTo(dataSource);
+
+    var originalFilterFactory = SourceModelWithTenant.createChangeFilter;
+    SourceModelWithTenant.createChangeFilter = function(since, modelFilter) {
+      var tenantProperty = TargetModelWithTenant.settings.tenantProperty;
+
+      var filter = originalFilterFactory.call(this, since, modelFilter);
+
+      if (modelFilter && modelFilter.where && modelFilter.where[tenantProperty]) {
+        filter.where.tenant = modelFilter.where[tenantProperty];
+      }
+      return filter;
+    };
+
+    SourceModelWithTenant.prototype.fillChangeTenantValue = function(change, cb) {
+      var model = TargetModelWithTenant;
+      var tenant = null;
+      var tenantProperty;
+      var inst = this;
+
+      if (model && model.settings && model.settings.tenantProperty) {
+        tenantProperty = model.settings.tenantProperty;
+      }
+
+      if (tenantProperty && inst && inst[tenantProperty]) {
+        tenant = inst[tenantProperty];
+      }
+
+      change.tenant = tenant;
+
+      cb(null);
+    };
+
+    test.startingCheckpoint = -1;
+  });
+
+  describe('Model.changes(since, filter, callback) - tenant variant', function() {
+    var tenant = '123';
+
+    it('Get changes since the given checkpoint for the given tenant', function(done) {
+      var test = this;
+      this.SourceModelWithTenant.create([{name: 'foo', tenantId: tenant}, {name: 'foo', tenantId: '456'}], function(err) {
+        if (err) return done(err);
+
+        setTimeout(function() {
+          test.SourceModelWithTenant.changes(test.startingCheckpoint, {where: {tenantId: tenant}}, function(err, changes) {
+            assert.equal(changes.length, 1);
+            assert.equal(changes[0].tenant, tenant);
```

Comment:
I believe this test would pass even without your changes applied. While it's ok to verify that the expected changes are returned, we also need to verify that correct queries were made against the datasource, i.e. that `tenantId` filter was applied when making the first `find` query on the `Change` model.